tests/raftstore/test_split_region.rs

```diff
@@ -249,3 +253,96 @@ fn test_server_delay_split_region() {
     let mut cluster = new_server_cluster(0, 3);
     test_delay_split_region(&mut cluster);
 }
+
+struct DropSnapshotFailure {
+    count: AtomicUsize,
+}
+
+impl Filter for DropSnapshotFailure {
+    fn before(&self, m: &RaftMessage) -> bool {
+        if m.get_message().get_msg_type() != raftpb::MessageType::MsgSnapStatus {
+            return false;
+        }
+
+        // ignore the first snapshot failure message
+        self.count.fetch_add(1, Ordering::SeqCst) < 1
+    }
+}
+
+fn test_split_overlap_snapshot<T: Simulator>(cluster: &mut Cluster<T>,
+                                             drop_snapshot_failure: bool) {
+    // We use three nodes 1, 2, 3 for this test.
+    cluster.bootstrap_region().expect("");
+    cluster.start();
+
+    // guarantee 1 is leader
+    cluster.transfer_leader(1, util::new_peer(1, 1));
+    cluster.must_put(b"k0", b"v0");
+    assert_eq!(cluster.leader_of_region(1), Some(util::new_peer(1, 1)));
+
+    let pd_client = cluster.pd_client.clone();
+
+    // isolate 3 for region 1.
+    cluster.hook_transport(IsolateRegionStore::new(1, 3));
+    cluster.must_put(b"k1", b"v1");
+
+    if drop_snapshot_failure {
+        // when region 1 split, node 1 will send snapshot to node 2 for region 2,
+        // but node 2 will returns a snapshot failure message. We should drop
```

Comment:
because the split log can be sent to node 2, but the region 2 snapshot can be, so the snapshot is overlapped, and we will return failure now.